indico/modules/users/client/js/react/components/ICSCalendarLink.jsx

```diff
@@ -0,0 +1,135 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Dropdown, Icon, Input, Label, Grid, Popup, Header} from 'semantic-ui-react';
+
+import signURL from 'indico-url:core.sign_url';
+
+import {Translate} from 'indico/react/i18n';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {snakifyKeys} from 'indico/utils/case';
+
+export default function ICSCalendarLink({endpoint, urlParams, options, ...restProps}) {
+  const [copied, setCopied] = useState(false);
+  const [option, setOption] = useState(null);
+  const [open, setOpen] = useState(false);
```

Comment:
do we need state and a controlled component for this? `Popup` can handle this on its own when not using the `open` prop

Reply:
We do, because we are opening the pop-up on the parent element instead of each option. Opening one per element would not be feasible since any click outside would close the dropdown menu (basically: a mess).